end-to-end-untargeted-metabolomics.Rmd

```diff
@@ -1,93 +1,744 @@
+```{r}
+#' get  data before alignment
+data_raw <- dropAdjustedRtime(data)
+```
+
+We can use this to compare the BPC before and after alignment
+
+```{r bpc before and after, echo=FALSE}
+
+#' Plot the BPC before and after alignment
+par(mfrow = c(2,1), mar = c(2, 1, 1, 0.5))
+plot(bpc_raw, peakType = "none", main = "BPC before alignment", col = paste0(col_phenotype[bpc$phenotype], 80))
+grid()
+legend("topright", col = col_phenotype,
+       legend = names(col_phenotype), lty = 1)
+plot(chromatogram(data, aggregationFun = "max", chromPeaks = "none"), main = "BPC after alignment", col = paste0(col_phenotype[bpc$phenotype], 80))
+grid()
+legend("topright", col = col_phenotype,
+       legend = names(col_phenotype), lty = 1)
+```
+
+
 - use subset alignment.
 
+Can also observe it in our ions of interest:
+
+```{r specific ion before and after, echo=FALSE}
+#' Similar comparison but this time filtering to see cystine
+par(mfrow = c(1, 2), mar = c(4, 4.5, 1, 0.5))
+old_eic_cystine <- chromatogram(data_raw, mz = cystine_mz + c(-0.05, 0.05), rt = c(205, 214))
+plot(old_eic_cystine, main = "Cystine before alignment")
+eic_cystine <- chromatogram(data,  mz = cystine_mz + c(-0.05, 0.05), rt = c(205, 214)) 
+plot(eic_cystine, "Cystine after alignment")
+```
+
+Little difference
+
+```{r echo=FALSE}
+#' Same for Met
+par(mfrow = c(1, 2), mar = c(4, 4.5, 1, 0.5))
+old_eic_Met <- chromatogram(data_raw, mz = Met_mz + c(-0.05, 0.05), 
+                            rt = c(175, 195)) 
+plot(old_eic_Met, main = "1- Methylhistidine before alignment" )
+eic_Met <- chromatogram(data,  mz = Met_mz + c(-0.05, 0.05), rt = c(175, 195))
+plot(eic_Met, main = "1- Methylhistidine after alignment" )
+```
+Our test ions did not display any significant differences between the raw and 
+adjusted data. 
+
+Use another ion as example...
+
 ## Correspondence
 
 - use *bad* settings (check maybe xcmsTutorials for examples)
 
+```{r}
+#' Default parameter for the grouping and apply them to the test ions BPC 
+param <- PeakDensityParam(sampleGroups = sampleData(data)$phenotype, bw = 30)
+plotChromPeakDensity(eic_cystine, param = param)
+plotChromPeakDensity(eic_Met, param = param)
+```
+
+- Good settings 
+```{r}
+#'Updating parameters
+param <- PeakDensityParam(sampleGroups = sampleData(data)$phenotype,
+                        minFraction = 1/3, binSize = 0.015, bw = 2)
+plotChromPeakDensity(eic_cystine, param = param)
+plotChromPeakDensity(eic_Met, param = param)
+```
+
+
+Now...
+
+```{r}
+#' Now apply to whole data
+data <- groupChromPeaks(data, param = param)
+```
+
+Test by checking an area with ions that are isomers of isoleucine and leucine 
+```{r}
+#' Extract chromatogram with signal for isomers leucine and isoleucine
+chr_test <- chromatogram(data, mz = 132.10191 + c(-0.01, 0.01),  
+                         rt = c(10, 180),aggregationFun = "max")
+plotChromPeakDensity(chr_test, simulate = FALSE)
+```
+Looks good 
```

Comment:
maybe you can also check it for 1-methylhistidine and 3-methylhistidine that you already used as an example above? that should be two close-by, separate peaks that get (hopefully) grouped into two separate features.

Reply:
ooh of course that's great I will try